app/src/main/java/com/example/androidtrainingproject/MainActivity.kt

```diff
@@ -37,6 +92,130 @@ fun Greeting(name: String, modifier: Modifier = Modifier) {
+    val hardcodedPassword = "1234"
+
+    Column(modifier = Modifier.padding(10.dp)) {
+        Text(text = "Log in", fontSize = 26.sp , modifier = Modifier
+            .padding(bottom = 10.dp)
+
+        )
+        TextField(
+            value = usernameField.value,
+            onValueChange = {
+                newValue -> usernameField.value = newValue
+            },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(bottom = 10.dp)
+                .border(BorderStroke(1.dp, Color.Gray), shape = RoundedCornerShape(3.dp)),
+            placeholder = { Text(text = "Email")}
+        )
+        TextField(
+            value = passwordField.value,
+            onValueChange = {
+                newValue -> passwordField.value = newValue
+            },
+            modifier = Modifier
+                .padding(bottom = 10.dp)
+                .fillMaxWidth()
+                .border(BorderStroke(1.dp, Color.Gray), shape = RoundedCornerShape(3.dp)),
+            placeholder = { Text(text = "Password")},
+            visualTransformation = if (isPasswordVisible.value) VisualTransformation.None else PasswordVisualTransformation(),
+            trailingIcon = {
+                IconButton(onClick = {
+                    isPasswordVisible.value = !isPasswordVisible.value
+                }) {
+                    if (isPasswordVisible.value) {
+                        Image(
+                            painter = painterResource(id = R.drawable.ic_visibility_on),
+                            contentDescription = "password visibility on",
+                            Modifier
+                                .height(36.dp)
+                                .width(32.dp)
+                        )
+                    } else {
+                        Image(
+                            painter = painterResource(id = R.drawable.ic_visibility_off),
+                            contentDescription = "password visibility off",
+                            Modifier
+                                .height(36.dp)
+                                .width(32.dp)
+                        )
+                    }
+                }
+            },
+        )
+        Button(
+            onClick = {
+                if (usernameField.value == hardcodedUsername && passwordField.value == hardcodedPassword) {
+                    navigateToProductDetails()
+                }
+                passwordField.value = ""
+            },
+            modifier = Modifier
+                .padding(horizontal = 20.dp)
+                .fillMaxWidth()
+        ) {
+            Text("Log in")
+        }
+    }
+}
+
+@Composable
+fun ProductDetails() {
+    Column(
+        Modifier
+            .fillMaxSize()
+            .padding(10.dp)) {
+        Text(
+            text = "Item",
+            textAlign = TextAlign.Center,
+            modifier = Modifier.fillMaxWidth()
+        )
+        Box(modifier = Modifier.fillMaxWidth()) {
```

Comment:
There's no reason to wrap a single element, you can do that in the composable's modifier.

Reply:
I wasn't able to center it without it, but now when I got back to it again I figured it out